pixel_clusterizer/clusterizer.py

```diff
@@ -182,73 +199,106 @@ def set_end_of_event_function(self, function):
         else:
             self.cluster_functions._end_of_event_function = function
 
-    def set_max_hits(self, value):
-        self._max_hits = value
-        self.hits_clustered = np.zeros(shape=(self._max_hits, ), dtype=np.dtype(self._hit_clustered_descr))
-        self.cluster = np.zeros(shape=(self._max_hits, ), dtype=np.dtype(self._cluster_descr))
+    def set_min_hit_charge(self, value):
+        ''' Charge values below this value will effectively ignore the hit.
+        Value has influence on clustering charge weighting.
+        '''
+        self._min_hit_charge = value
 
     def set_max_hit_charge(self, value):
-        self._max_cluster_hit_charge = value
-
-    def set_x_cluster_distance(self, value):
-        self._x_cluster_distance = value
-
-    def set_y_cluster_distance(self, value):
-        self._y_cluster_distance = value
+        ''' Charge values above this value will effectively ignore the hit.
+        Value of None or 0 will deactivate this feature.
+        '''
+        if value is None:
+            value = 0
+        self._max_hit_charge = value
+
+    def set_column_cluster_distance(self, value):
+        ''' Setting up max. column cluster distance.
+        '''
+        self._column_cluster_distance = value
+
+    def set_row_cluster_distance(self, value):
+        ''' Setting up max. row cluster distance.
+        '''
+        self._row_cluster_distance = value
 
     def set_frame_cluster_distance(self, value):
+        ''' Setting up max. frame cluster distance.
+        '''
         self._frame_cluster_distance = value
 
-    def set_max_cluster_hits(self, value):
-        self._max_cluster_hits = value
-
-    def ignore_same_hits(self, value):  # Ignore same hit in an event for clustering
+    def ignore_same_hits(self, value):
+        ''' Whether a duplicate hit in the event with the same column and row is ignored or not.
+        '''
         self._ignore_same_hits = value
 
-    def create_cluster_hit_info_array(self, value=True):  # TODO: do not create cluster hit info of false to save time
-        self._create_cluster_hit_info_array = value
+    def cluster_hits(self, hits, noisy_pixels=None, disabled_pixels=None):
+        ''' Cluster given hit array.
+
+        The noisy_pixels and disabled_pixels parameters are iterables of column/row index pairs, e.g. [[column_1, row_1], [column_2, row_2], ...].
+        The noisy_pixels parameter allows for removing clusters that solely consists of noisy pixels.
+        The disabled_pixels parameter allows for ignoring pixels.
```

Comment:
It is not too clear to me what the behaviour is, thus I make some 1D examples with cluster distance 0 (`hit hit nohit hit` gives 2 clusters with cluster size 2/1). Can you tell me if these cases would give one or two clusters and which cluster size?

-  `hit hit disabled hit` (gives I guess 1 cluster with cluste size 3?)
-  `hit hit disabled disabled hit`
-  `hit hit noise hit`
-  `hit hit noise noise hit`
-  `hit hit noise disabled hit`
-  `hit hit noise disabled noise disabled hit`
-  `noise noise`
-  `disabled noise noise`